subsys/bluetooth/audio/bap_broadcast_source.c

```diff
@@ -595,6 +595,7 @@ static bool valid_broadcast_source_param(const struct bt_bap_broadcast_source_pa
 			}
 
 			CHECKIF(subgroup_param->codec_cfg->id == BT_HCI_CODING_FORMAT_LC3 &&
+				stream_param->data_len > 0 &&
```

Comment:
```suggestion
				stream_param->data != NULL &&
```

I would move this check before `subgroup_param->codec_cfg->id == BT_HCI_CODING_FORMAT_LC3` and check the `data` pointer instead of the length, as the length is not the issue, but the pointer is.